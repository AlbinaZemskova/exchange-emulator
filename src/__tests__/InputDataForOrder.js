import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallowToJson } from 'enzyme-to-json';
import {
  InputDataForOrder,
  mapStateToProps,
  mapDispatchToProps,
} from '../InputDataForOrder';
import 'jest-prop-type-error';

Enzyme.configure({ adapter: new Adapter() });

describe('<InputDataForOrder />', () => {
  let wrapper;
  let addOrderForBuy;
  let addOrderForSell;
  let buyOrders;
  let price;
  let count;
  let event;

  beforeEach(() => {
    event = {
      target: {
        reset: jest.fn(),
      },
      preventDefault: jest.fn(),
    };

    addOrderForSell = jest.fn();
    addOrderForBuy = jest.fn();

    buyOrders = [
      {
        price: 1000,
        count: 200,
      },
      {
        price: 2000,
        count: 150,
      },
    ];

    wrapper = shallow(
      <InputDataForOrder
        addOrderForBuy={addOrderForBuy}
        addOrderForSell={addOrderForSell}
      />,
    );
  });

  describe('InputDataForOrder component', () => {
    describe('render', () => {
      it('renders without crashing', () => {
        expect(shallowToJson(wrapper)).toMatchSnapshot();
      });

      it('checking correct proptypes', () => {
        expect(() => (
          <InputDataForOrder
            buyOrders={[
              {
                price: '1',
                count: '2',
              },
            ]}
            sellOrders={[
              {
                price: '1',
                count: '2',
              },
            ]}
          />
        )).toThrow();
      });
    });

    describe('methods', () => {
      it('handles getting order for buy', () => {
        price = 1000;
        count = 100;

        wrapper.instance().handleOrderOfBuy(price, count, event);
        expect(addOrderForBuy).toHaveBeenCalledWith({ price, count });
      });

      it('handles getting order for sell', () => {
        price = 2000;
        count = 150;

        wrapper.instance().handleOrderOfSell(price, count, event);
        expect(addOrderForSell).toHaveBeenCalledWith({ price, count });
      });
    });
  });

  describe('mapStateToProps', () => {
    it('should return an object with the arrays of objects', () => {
      const mockState = {
        buyOrders: {
          buyOrders,
        },
      };

      const expected = {
        buyOrders,
      };

      const mappedProps = mapStateToProps(mockState);

      expect(mappedProps).toEqual(expected);
    });
  });

  describe('mapDispatchToProps', () => {
    const dispatch = jest.fn();

    const actions = mapDispatchToProps(dispatch);
    actions.addOrderForBuy();
    expect(dispatch).toHaveBeenCalled();
  });
});
