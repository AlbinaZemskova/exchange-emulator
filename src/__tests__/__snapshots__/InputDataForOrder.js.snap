// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputDataForOrder /> InputDataForOrder component render renders without crashing 1`] = `
<Fragment>
  <Container
    tag="div"
  >
    <Row
      tag="div"
    >
      <Col
        lg="6"
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <h3>
          Left your order for buy
        </h3>
        <Form
          handleSubmit={[Function]}
        />
      </Col>
      <Col
        lg="6"
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <h3>
          Left your order for sell
        </h3>
        <Form
          handleSubmit={[Function]}
        />
      </Col>
    </Row>
  </Container>
  <hr />
</Fragment>
`;
